lib/dialects/abstract/query-generator.js

```diff
@@ -1513,7 +1513,16 @@ const QueryGenerator = {
     const primaryKeysTarget = association.target.primaryKeyAttributes;
     const tableTarget = includeAs.internalAs;
     const identTarget = association.foreignIdentifierField;
-    const attrTarget = association.target.rawAttributes[primaryKeysTarget[0]].field || primaryKeysTarget[0];
+
+    // Determine the attrTarget, if there is a NonPrimaryBelongsTo in our association
+    // we need to use the targetPrimaryKeyAttribute to ensure we do not default to the standard
+    // primary key.
+    let attrTarget;
```

Comment:
May be both `attrTarget` and `attrSource` should be exposed by associations class

Reply:
I don't understand this request/comment.

Reply:
@sushantdhiman, would you clarify this request?